src/features/SaveData/components/FileUpload.tsx

```diff
@@ -0,0 +1,50 @@
+import { useRef, useState } from 'react'
+
+interface FileUploadProps {
+  onUpload: (files: FileList) => void | Promise<void>
+}
+
+export default function FileUpload({ onUpload }: FileUploadProps) {
+  const fileUploadElement = useRef<HTMLInputElement>(null)
+  const [over, setOver] = useState(false)
+  return (
+    <div
+      data-over={over}
+      className='data-[over=true]:p-64 data-[over=true]:border-4 border-dotted inline-block'
+      onDrop={e => {
+        e.preventDefault()
+        e.stopPropagation()
+        setOver(false)
+        onUpload(e.dataTransfer.files)
+      }}
+      onDragOver={e => {
+        e.preventDefault()
+        e.stopPropagation()
+        setOver(true)
+      }}
+      onDragLeave={e => {
+        e.preventDefault()
+        e.stopPropagation()
+        setOver(false)
+      }}
+    >
+      <input
+        ref={fileUploadElement}
+        className='hidden'
+        type='file'
+        accept='.csv'
+        multiple
+        onChange={e => {
+          const target = e.target as HTMLInputElement & { files: FileList }
+          onUpload(target.files)
+        }}
+      />
+      <button
+        className='border p-2 rounded hover:bg-gray-100 transition-all'
+        onClick={() => fileUploadElement.current?.click()}
+      >
+        Upload File Here
```

Comment:
I'm assuming this is something we'll care about in the future, but should there be any further instruction? For example: "Make sure your file follows this format X (or even an image)". Would it be helpful if we add some dummy place holder text for now so we don't forget?

Reply:
Currently, I just put this element on the home page for testing reasons, but this feature isn't really in the scope of this PR. I think this issue might be better addressed in a pr specific for creating a page/some dialogs for uploading files.

There's still a bunch if functionality that needs to be implemented for that (e.g. check if uploaded files meet correct format, push file data to Dexie, create clear visual language around download/upload workflow)

I made a commit that removes the button from the HomePage and I'll add this as a task to our scrum board, but I think we should push off making decisions on this feature until it's own PR.